### PR TITLE
Better error message for invalid directory/file path in config file exceptions

### LINKED ISSUE #7111: Bug: psalm.xml cannot be loaded if a non-existent directory is specified
I just had this strange error 

>Problem parsing /var/www/html/psalm.xml:
  Could not resolve config path to /var/www/html/

thrown from [here](https://github.com/vimeo/psalm/blob/master/src/Psalm/Config/FileFilter.php#L175).

The cause was this snippet in my `psalm.xml`:

```xml
        <RedundantConditionGivenDocblockType>
            <errorLevel type="suppress">
                <directory name="bar/"/>
            </errorLevel>
        </RedundantConditionGivenDocblockType>
```

 I just had moved `bar` to `foo/bar`. So it could not resolve `bar` (`realpath` in Line 168 returned false). That was hard to find because the error was not very informative. Maybe we could change that so that I can see the real mistake a bit more clearly? 😅

### PR DESCRIPTION
**Pull request by SMAtaurRahman**
Fixes #7111 

Let me know if you have any suggestion on improving error messages.

Note: I wasn't able to re-produce the exceptions thrown in line 153 & 258. So I didn't write any tests for them. If you know how, please let me know.

Thanks.

### CODE REVIEW

File: src/Psalm/Config/FileFilter.php [hunk 1]
```diff
@@ -246,16 +256,17 @@ public static function loadFromArray(
                         }
 
                         throw new ConfigException(
-                            'Could not resolve config path to ' . $base_dir . DIRECTORY_SEPARATOR .
-                            $file_path
+                            'Could not resolve file path to ' . $prospective_file_path . PHP_EOL .
+                            'The path is defined in config file. Does this file exist?'
```
Comment (orklah) [inline src/Psalm/Config/FileFilter.php:260]: I'd have said something like 'This path is defined in your config file. Please make sure this file exists.'

Comment (SMAtaurRahman) [reply src/Psalm/Config/FileFilter.php:260]: Yeah, that sounds better.

**Comment by SMAtaurRahman** [comment]
Ugh, broken windows tests again... checking......

File: src/Psalm/Config/FileFilter.php (hunk 1 above)
Comment (SMAtaurRahman) [reply src/Psalm/Config/FileFilter.php:260]: Could we also add another line with a suggestions to use `allowMissingFiles` ?
Something like, `or use allowMissingFiles="true" to ignore non-existent  files`.

Comment (orklah) [reply src/Psalm/Config/FileFilter.php:260]: oh yeah, great idea

**Comment by orklah** [comment]
@SMAtaurRahman do you need help on that? :)

**Comment by SMAtaurRahman** [comment]
@orklah Yes, please, I would appreciate the help. :)

**Comment by orklah** [comment]
Do you mind rebasing on 4.x first? I'll clone your branch and do some tests after

**Comment by orklah** [comment]
make sure to rebase on 4.x and not on master (master is the next psalm 5)

File: tests/Config/ConfigTest.php [hunk 2]
```diff
@@ -1549,4 +1549,128 @@ public function testStrictTypesForFileReporting(): void
...
+            'This path is defined in your config file. Please make sure this directory exists.' . PHP_EOL .
+            'Or add allowMissingFiles="true" to your config file to ignore non-existent files.');
+
+        $this->project_analyzer = $this->getProjectAnalyzerWithConfig(
+            Config::loadFromXML(
+                (string) getcwd() . DIRECTORY_SEPARATOR,
+                '<?xml version="1.0"?>
+                <psalm>
+                    <projectFiles>
+                        <directory name="src2/*" />
```
Comment (orklah) [inline tests/Config/ConfigTest.php:1588]: use DIRECTORY_SEPARATOR here instead of /

File: tests/Config/ConfigTest.php [hunk 3]
```diff
@@ -1549,4 +1549,128 @@ public function testStrictTypesForFileReporting(): void
...
+            'This path is defined in your config file. Please make sure this directory exists.' . PHP_EOL .
+            'Or add allowMissingFiles="true" to your config file to ignore non-existent files.');
+
+        $this->project_analyzer = $this->getProjectAnalyzerWithConfig(
+            Config::loadFromXML(
+                (string) getcwd(),
+                '<?xml version="1.0"?>
+                <psalm>
+                    <projectFiles>
+                        <directory name="src/Psalm/Config/*" />
```
Comment (orklah) [inline tests/Config/ConfigTest.php:1609]: same here, and tests pass on windows afterwards

**Comment by orklah** [comment]
Putting this in draft, please ping me once it's fixed

**Comment by weirdan** [comment]
4.x branch is closed now as we prepare for the 5.0 release. Please target the master branch instead.

**Comment by orklah** [comment]
Closing as this targets 4.x which is closed. Please reopen on master going forward

**Pull request closed without merging**